core/tests/protractor_utils/StoryEditorPage.js

```diff
@@ -139,6 +198,144 @@ var StoryEditorPage = function() {
     closeSaveModalButton.click();
     waitFor.pageToFullyLoad();
   };
+
+  this.expectSaveStoryDisabled = function() {
+    return expect(
+      saveStoryButton.getAttribute('disabled')).toEqual('true');
+  };
+
+  this.expectDisplayUnreachableChapter = function() {
+    return expect(disconnectedChapterWarning.isPresent()).toBe(true);
+  };
+
+  this.setChapterExplorationId = function(explorationId) {
+    waitFor.visibilityOf(
+      explorationIdInput,
+      'ExplorationIdInput takes too long to be visible'
+    );
+
+    explorationIdInput.sendKeys(explorationId);
+    waitFor.elementToBeClickable(
+      explorationIdSaveButton,
+      'ExplorationIdSaveButton takes too long to be visible'
+    );
+    explorationIdSaveButton.click();
+  };
+
+  this.changeNodeOutline = function(richTextInstructions) {
+    var editor = forms.RichTextEditor(
+      nodeOutlineEditor);
+    editor.clear();
+    richTextInstructions(editor);
+    nodeOutlineSaveButton.click();
+  };
+
+  this.navigateToChapterByIndex = function(index) {
+    chapterTitles.then(function(elements) {
+      elements[index].click();
+    });
+  };
+
+  this.expectExplorationIdAlreadyExistWarning = function() {
+    var warningToast = element(by.css('.toast-message'));
+    waitFor.visibilityOf(
+      warningToast,
+      'warningToast takes too long to be visible.');
+    expect(warningToast.getText()).toEqual(
+      'The given exploration already exists in the story.');
+    var closeToastButton = element(by.css('.toast-close-button'));
+    waitFor.elementToBeClickable(
+      closeToastButton,
+      'closeToastButton takes too long to be clickable.');
+    closeToastButton.click();
+  };
+
+  this.getSelectSkillModal = function() {
+    waitFor.visibilityOf(
+      selectSkillModalHeader,
+      'selectSkillModalHeader takes too long to be visible.');
+    return {
+      _searchSkillByName: function(name) {
+        waitFor.visibilityOf(
+          skillNameInputField,
+          'skillNameInputField takes too long to be visible');
+        skillNameInputField.sendKeys(name);
+      },
+
+      _selectSkillBasedOnIndex: function(index) {
+        skillListItems.then(function(elements) {
+          var selectedSkill = elements[index];
+          waitFor.elementToBeClickable(
+            selectedSkill,
+            'selectedSkill takes too long to be clickable.'
+          );
+          selectedSkill.click();
+        });
+      },
+
+      selectSkill: function(name) {
+        this._searchSkillByName(name);
+        this._selectSkillBasedOnIndex(0);
+        var doneButton = element(by.css('.btn-success'));
```

Comment:
We prefer finding elements using HTML classes of the form `protractor-test-*` that way we don't accidentally break the references. I'm not sure if this means this line should change. Awaiting guidance from @nithusha21.

Reply:
@cuichenli I've heard from @nithusha21 that this should indeed use an HTML class of the form `protractor-test-*`. I think (though haven't checked) that there are other places in the code where you get an element by a class other than `protractor-test-*`, which should also be updated